### ISSUE TITLE
.hbjson too large for 32-bit reader

### ISSUE THREAD

**Issue by rangisutton** [sideeffects/HoudiniNiagara#41]
Doing stress tests for a show and I'm hitting limits on the numbers/length of the cache I'm importing ( not a real time project.)

On import it fails with this message: 

LogStreaming: Error: File 'XXXX.hbjson' is too large for 32-bit reader (50484506577), use TArray64.

Saw another user with same issue on the sidefx forum.

I can supply an example cache but I'm sure anything large enough will trigger this message.

UE5.1.1, win 10

Thanks!

**Comment by felixschuch** [comment]
I am having the same Issue with large caches.
Also noticed that writing .hbjsons in Houdini gets really slow with increasing particle counts. Any chance this could work with sequences? Which would also solve the problem with large files.

**Comment by b0oing** [comment]
having the same issue on my end. i did try to import the .hbjson on 5.0.3 and it worked. there has to be something wrong with the importing method on the current version of 5.1.1.
the best i would suggest would be to create your scene in an older version of UE and reimport the niagara files directly. let's hope it gets fixed soon.

**Comment by ktk-ring** [comment]
Have you found a solution to this problem? I have same issue on my project. I have imported large hbjson file, but the Output Log says "File XXX.hbjson is too large for 32-bit reader, use TArray64"

I also did try to import it on 5.0.3 but it didn't work either. Why this plugin import particle caches into 32-bit array? I cannot understand why...

**Comment by rangisutton** [comment]
We never found a solve that allowed us to use niagra caches. We went with a different approach using PCG in 5.3 that worked for our use case ( large animated set of instanced geo  ).

**Comment by wildparky** [comment]
> We never found a solve that allowed us to use niagra caches. We went with a different approach using PCG in 5.3 that worked for our use case ( large animated set of instanced geo ).

Can you explain your use case in more detail?

**Comment by matteo-collina** [comment]
I am running into the same issue. Did someone find a solution to this?

**Comment by brydlst** [comment]
We have the same issue in our team.
Looking at the source code of the plugin it seems that to fix this issue it would unfortunately require to revrite quite a bit of the plugin to use TArray64 instead.

**Comment by matteo-collina** [comment]
I have still not found a solution. I have opened a ticket to the SideFX support but they told me they will get back to me when an updated will be available…which is probably not going to happen anytime soon. So annoying!

**Comment by Rowquino** [comment]
Also waiting for a solution on this, has anyone figured out a way to convert the plugin to TArray64?

**Comment by matteo-collina** [comment]
No. Considering they didn't make much progress over the last 7 months, I think they are just maintaining this tool without adding new features, unfortunately.

**Comment by brydlstepan** [comment]
Managed to rework the plugin to solve this issue until an official solution is released. If anyone is interested:
https://github.com/brydlstepan/HoudiniNiagara/releases/tag/v2.7.4a

Important: this version is **not backward- or forward-compatible** with the original plugin .hbjson files.

**Comment by dpernuit** [comment]
Hi,
Sorry for the late reply - I've finally got the time to look into fixing this.
To be fair - the plugin was never designed with the intention of supporting massive data caches - and that is something we used to advise against when we released the plugin.
That being said -  I can see the benefits of supporting such large caches for non-realtime workflows.
I'll let you all know when a fix for this is in.

**Comment by dpernuit** [comment]
Hi,
That issue should now be fixed with the new release that I pushed (2.8).
Files over 2GB will now import properly - there is still however a limit in file size (~4GB) - as Niagara only uses 32 bits indexing internal for its buffer.
If you want to import larger files - I recommend splitting the point caches to multiple files.
